build command issue

Since meson branch merge, the build command need probably updated in https://github.com/Nic30/hdlConvertor/blob/master/README.md

When I use:
```
pip3 install .
```

The installation failed with:
```
      ccache c++ -Isrc/libhdlConvertor_cpp_static.a.p -Isrc -I../src -Iinclude -I../include -I../subprojects/antlr4-runtime/runtime/Cpp/runtime/src -I../subprojects/antlr4-runtime/runtime/Cpp/runtime/src/atn -I../subprojects/antlr4-runtime/runtime/Cpp/runtime/src/dfa -I../subprojects/antlr4-runtime/runtime/Cpp/runtime/src/internal -I../subprojects/antlr4-runtime/runtime/Cpp/runtime/src/misc -I../subprojects/antlr4-runtime/runtime/Cpp/runtime/src/support -I../subprojects/antlr4-runtime/runtime/Cpp/runtime/src/tree -I../subprojects/antlr4-runtime/runtime/Cpp/runtime/src/tree/pattern -I../subprojects/antlr4-runtime/runtime/Cpp/runtime/src/tree/xpath -Isubprojects/antlr4-runtime -I../subprojects/antlr4-runtime -fdiagnostics-color=always -DNDEBUG -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c++17 -O3 -fPIC -Wall -pedantic -W -Wno-attributes -MD -MQ src/libhdlConvertor_cpp_static.a.p/hdlConvertor.cpp.o -MF src/libhdlConvertor_cpp_static.a.p/hdlConvertor.cpp.o.d -o src/libhdlConvertor_cpp_static.a.p/hdlConvertor.cpp.o -c ../src/hdlConvertor.cpp
      In file included from ../include/hdlConvertor/hdlConvertor.h:14,
                       from ../src/hdlConvertor.cpp:1:
      ../include/hdlConvertor/verilogPreproc/verilogPreproc.h:13:10: fatal error: hdlConvertor/verilogPreproc/verilogPreprocParser/verilogPreprocLexer.h: No such file or directory
         13 | #include <hdlConvertor/verilogPreproc/verilogPreprocParser/verilogPreprocLexer.h>
            |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      compilation terminated.
      [173/248] Generating verilogPreproc_src_gen with a custom command
      warning(179): verilogPreprocLexer.g4:369:58: incompatible commands type and skip
      [174/248] Generating vhdl_src_gen with a custom command
      [175/248] Generating sv2017_src_gen with a custom command
      ninja: build stopped: subcommand failed.
``` 

Apparently files C++ files from the grammar are not generated out of the box, and are missing for the compilation of source code.

Minimal command that works for me is:
```
pip3 install . -Ccompile-args="-j1"
```
Or to compile with debug symbol
```
pip -v  install -e . -Csetup-args="-Dbuildtype=debug" -Ccompile-args="--verbose" -Cbuild-dir=build-dbg -Ccompile-args="-j1"
```

[log.zip](https://github.com/user-attachments/files/20891049/log.zip)

```
      [176/248] Compiling C++ object src/libhdlConvertor_cpp_static.a.p/hdlConvertor.cpp.o
      FAILED: src/libhdlConvertor_cpp_static.a.p/hdlConvertor.cpp.o
      ccache c++ -Isrc/libhdlConvertor_cpp_static.a.p -Isrc -I../src -Iinclude -I../include -I../subprojects/antlr4-runtime/runtime/Cpp/runtime/src -I../subprojects/antlr4-runtime/runtime/Cpp/runtime/src/atn -I../subprojects/antlr4-runtime/runtime/Cpp/runtime/src/dfa -I../subprojects/antlr4-runtime/runtime/Cpp/runtime/src/internal -I../subprojects/antlr4-runtime/runtime/Cpp/runtime/src/misc -I../subprojects/antlr4-runtime/runtime/Cpp/runtime/src/support -I../subprojects/antlr4-runtime/runtime/Cpp/runtime/src/tree -I../subprojects/antlr4-runtime/runtime/Cpp/runtime/src/tree/pattern -I../subprojects/antlr4-runtime/runtime/Cpp/runtime/src/tree/xpath -Isubprojects/antlr4-runtime -I../subprojects/antlr4-runtime -fdiagnostics-color=always -DNDEBUG -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c++17 -O3 -fPIC -Wall -pedantic -W -Wno-attributes -MD -MQ src/libhdlConvertor_cpp_static.a.p/hdlConvertor.cpp.o -MF src/libhdlConvertor_cpp_static.a.p/hdlConvertor.cpp.o.d -o src/libhdlConvertor_cpp_static.a.p/hdlConvertor.cpp.o -c ../src/hdlConvertor.cpp
      In file included from ../include/hdlConvertor/hdlConvertor.h:14,
                       from ../src/hdlConvertor.cpp:1:
      ../include/hdlConvertor/verilogPreproc/verilogPreproc.h:13:10: fatal error: hdlConvertor/verilogPreproc/verilogPreprocParser/verilogPreprocLexer.h: No such file or directory
         13 | #include <hdlConvertor/verilogPreproc/verilogPreprocParser/verilogPreprocLexer.h>
            |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      compilation terminated.
      [177/248] Generating verilogPreproc_src_gen with a custom command
      warning(179): verilogPreprocLexer.g4:369:58: incompatible commands type and skip
      [178/248] Compiling Cython source /home/tom/prog/git/hdlConvertor/hdlConvertor/_hdlConvertor.pyx
      [179/248] Generating vhdl_src_gen with a custom command
      [180/248] Generating sv2017_src_gen with a custom command
      ninja: build stopped: subcommand failed.
      [end of output]
```

It looks like dependencies are not address in correct order : meson try to compile C++ before custom command that generate antlr parser and lexer are generated.

--- a/meson.build
+++ b/meson.build
@@ -89,6 +89,8 @@ mainIncludeDir = 'include/'
 mainIncludeDir_inc = include_directories(mainIncludeDir)
 all_parser_libs = []
 all_parser_deps = []
+all_parser_header_targets = []
+
 foreach parserTuple : ANTLR_PARSERS
 	name = parserTuple[0]
 	folder_name = parserTuple[1]
@@ -147,7 +149,11 @@ foreach parserTuple : ANTLR_PARSERS
 		endif
 	endforeach
 	genIncludeDir = join_paths('include', 'hdlConvertor', folder_name, f'@name@Parser')
-  	subdir(genIncludeDir)
+	subdir(genIncludeDir)
+
+	# Add the header copy target to our list of dependencies
+	all_parser_header_targets += antrl_parser_headers
+
 	parser_cpp_src = run_command(python_prog, 'utils/rec_glob.py', join_paths('src', folder_name), '*.cpp', check:true)\
     	.stdout().strip().split('\n')
 	

--- a/src/meson.build
+++ b/src/meson.build
@@ -15,7 +15,7 @@ hdlConvertor_core_src += run_command(py3, '../utils/rec_glob.py', './baseHdlPars
 	.stdout().strip().split('\n')
 
 hdlConvertor_core_static_lib = static_library('hdlConvertor_core_static',
-    hdlConvertor_core_src,
+    hdlConvertor_core_src + all_parser_header_targets,
     cpp_args: EXTRA_CXX_FLAGS,
     dependencies: [antlr4_cpp_dep, all_parser_deps],
     include_directories: [mainIncludeDir_inc],
@@ -32,7 +32,7 @@ if get_option('python_package')
 	# [note] hdlConvertor_core_static and hdlConvertor_cpp_static has to be separated
 	#        otherwise it is not possible to link this library
 	hdlConvertor_cpp_static = static_library('hdlConvertor_cpp_static',
-    	hdlConvertor_cpp_src,
+    	hdlConvertor_cpp_src + all_parser_header_targets,
 	    cpp_args: EXTRA_CXX_FLAGS,
 	    dependencies: [antlr4_cpp_dep, all_parser_deps],
 	    include_directories: [mainIncludeDir_inc],
@@ -42,7 +42,7 @@ if get_option('python_package')
 else
 	# add dummy file to make cmake happy
 	hdlConvertor_cpp_shared = shared_library('hdlConvertor_cpp_shared',
-		hdlConvertor_cpp_src,
+		hdlConvertor_cpp_src + all_parser_header_targets,
 	 	cpp_args: EXTRA_CXX_FLAGS,
 	    dependencies: [antlr4_cpp_dep, ] + all_parser_deps,
 	    include_directories: [mainIncludeDir_inc],

